components/core/src/clp/ffi/ir_stream/byteswap.hpp

```diff
@@ -3,16 +3,36 @@
 
 #if defined(__APPLE__)
     #include <libkern/OSByteOrder.h>
+
+    // TODO: Improve byte swapping implementation by replacing macros with a templated function.
+    // Ideally, we should implement a general, type-safe, templated byte swap function where the
+    // input size is inferred from the type. This would:
+    // - Improve usability by removing the need to specify the size explicitly.
+    // - Encapsulate platform-specific details, improving portability and maintainability.
+
+    // NOTE: Currently, we disable the following two checks for the following reasons:
+    // - We're mirroring the GNU `byteswap.h` method names (e.g., `bswap_16`), so we're using
+    //   lowercase names.
+    // - We're using macros as wrappers around platform-specific byte swap functions because we
+    //   don't yet have a proper templated implementation.
```

Comment:
```suggestion
    // NOTE:
    // - We disable `cppcoreguidelines-macro-usage` since we're using macros as wrappers around
    //   platform-specific byte swap functions, and we don't yet have a proper templated
    //   implementation.
    // - We disable `readability-identifier-naming` since we're mirroring the GNU `byteswap.h`
    //   method names (e.g., `bswap_16`) which are lowercase.
```